### ISSUE TITLE
pip install autotrain-advanced failed

### ISSUE THREAD

**Issue by pankajarm** [huggingface/autotrain-advanced#143]
Hi, first of all thanks for this amazing library, really appreciate the efforts behind it.

However, I am seeing pip install autotrain-advanced failed, I tried to install it on Paperspace [gradient,](url) A100 80GB & A6000 48GB machine few time, same issue
Here is the environment details:

lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.5 LTS
Release:        20.04
Codename:       focal

python --version
Python 3.9.16

nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Tue_Mar__8_18:18:20_PST_2022
Cuda compilation tools, release 11.6, V11.6.124
Build cuda_11.6.r11.6/compiler.31057947_0

Here is the detail log:

pip install autotrain-advanced
Collecting autotrain-advanced
  Downloading autotrain_advanced-0.5.5-py3-none-any.whl (68 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 68.1/68.1 kB 10.8 MB/s eta 0:00:00
Collecting boto3==1.16.49
  Downloading boto3-1.16.49-py2.py3-none-any.whl (130 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 130.2/130.2 kB 37.6 MB/s eta 0:00:00
Collecting codecarbon==2.2.3
  Downloading codecarbon-2.2.3-py3-none-any.whl (174 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 174.1/174.1 kB 50.8 MB/s eta 0:00:00
Collecting sacrebleu==2.0.0
  Downloading sacrebleu-2.0.0-py3-none-any.whl (90 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 90.7/90.7 kB 32.4 MB/s eta 0:00:00
Requirement already satisfied: pyyaml==5.4.1 in /usr/local/lib/python3.9/dist-packages (from autotrain-advanced) (5.4.1)
Collecting sacremoses==0.0.53
  Downloading sacremoses-0.0.53.tar.gz (880 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 880.6/880.6 kB 91.3 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting librosa==0.8.0
  Downloading librosa-0.8.0.tar.gz (183 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 183.9/183.9 kB 48.6 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting trl
  Downloading trl-0.4.6-py3-none-any.whl (75 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 75.7/75.7 kB 29.5 MB/s eta 0:00:00
Collecting psycopg2==2.8.6
  Downloading psycopg2-2.8.6.tar.gz (383 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 383.8/383.8 kB 73.9 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [25 lines of output]
      /usr/local/lib/python3.9/dist-packages/setuptools/config/setupcfg.py:515: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
        warnings.warn(msg, warning_class)
      running egg_info
      creating /tmp/pip-pip-egg-info-wp4amrbz/psycopg2.egg-info
      writing /tmp/pip-pip-egg-info-wp4amrbz/psycopg2.egg-info/PKG-INFO
      writing dependency_links to /tmp/pip-pip-egg-info-wp4amrbz/psycopg2.egg-info/dependency_links.txt
      writing top-level names to /tmp/pip-pip-egg-info-wp4amrbz/psycopg2.egg-info/top_level.txt
      writing manifest file '/tmp/pip-pip-egg-info-wp4amrbz/psycopg2.egg-info/SOURCES.txt'
      
      Error: pg_config executable not found.
      
      pg_config is required to build psycopg2 from source.  Please add the directory
      containing pg_config to the $PATH or specify the full executable path with the
      option:
      
          python setup.py build_ext --pg-config /path/to/pg_config build ...
      
      or with the pg_config option in 'setup.cfg'.
      
      If you prefer to avoid building psycopg2 from source, please install the PyPI
      'psycopg2-binary' package instead.
      
      For further information please check the 'doc/src/install.rst' file (also at
      <https://www.psycopg.org/docs/install.html>).
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.

**Comment by stavsap** [comment]
this helped me:

sudo apt-get install postgresql python3-dev
sudo apt-get install --reinstall libpq-dev -y
pip install psycopg2
pip install psycopg2-binary

**Comment by stavsap** [comment]
i was able to install with this https://github.com/huggingface/autotrain-advanced/issues/144#issuecomment-1633729207

**Comment by pankajarm** [comment]
> 

thank a lot @stavsap for this, really appreciated your help. 

Not sure you were recommending these libraries on paperspace gradient, but when I tried this on it, i hit the wall on psycopg2, so yeah again blocked on installing this amazing library on paperspace gradient or google collab, 

here is the detail log:

root@nbgrw8iu5c:/notebooks# pip install psycopg2
Collecting psycopg2
  Downloading psycopg2-2.9.6.tar.gz (383 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 384.0/384.0 kB 25.1 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [25 lines of output]
      /usr/local/lib/python3.9/dist-packages/setuptools/config/setupcfg.py:515: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
        warnings.warn(msg, warning_class)
      running egg_info
      creating /tmp/pip-pip-egg-info-5r_jcysb/psycopg2.egg-info
      writing /tmp/pip-pip-egg-info-5r_jcysb/psycopg2.egg-info/PKG-INFO
      writing dependency_links to /tmp/pip-pip-egg-info-5r_jcysb/psycopg2.egg-info/dependency_links.txt
      writing top-level names to /tmp/pip-pip-egg-info-5r_jcysb/psycopg2.egg-info/top_level.txt
      writing manifest file '/tmp/pip-pip-egg-info-5r_jcysb/psycopg2.egg-info/SOURCES.txt'
      
      Error: pg_config executable not found.
      
      pg_config is required to build psycopg2 from source.  Please add the directory
      containing pg_config to the $PATH or specify the full executable path with the
      option:
      
          python setup.py build_ext --pg-config /path/to/pg_config build ...
      
      or with the pg_config option in 'setup.cfg'.
      
      If you prefer to avoid building psycopg2 from source, please install the PyPI
      'psycopg2-binary' package instead.
      
      For further information please check the 'doc/src/install.rst' file (also at
      <https://www.psycopg.org/docs/install.html>).
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.

**Comment by stavsap** [comment]
try to run this 

sudo apt-get install postgresql libpq-dev -y

before installing psycopg2

**Comment by pankajarm** [comment]
yeah I did that, my bad, when I mentioned I hit wall, I mean, i did already install postgresql, python3-dev,  and reinstall  libpq-dev -y

sudo apt-get install postgresql python3-dev => DONE
sudo apt-get install --reinstall libpq-dev -y => DONE
pip install psycopg2 => ISSUE which i mentioned
pip install psycopg2-binary => need psycopg2 first

Were you able to run this on paperspace gradient successfully? if yes, can you please paste screenshot. I saw your post about ubuntu WSL on windows, but on paperspace gradient or collab I am not able to find work around to install this library. 
 
Again thanks for taking time to respond, much appreciated.

**Comment by abhishekkrthakur** [comment]
ill make the requirements light in next release today

**Comment by stavsap** [comment]
I didn't try on paperspace gradient or colab.

reverified install on Ubuntu 22.04 WSL

the following install should work.
`
sudo apt install cmake -y
sudo apt-get install postgresql python3-dev -y
sudo apt-get install --reinstall libpq-dev -y
sudo apt install libjpeg-dev zlib1g-dev -y

conda create -n autotrain python=3.9.0 -y
conda activate autotrain
conda install cudatoolkit -y
conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia -y
conda install -c conda-forge hdbscan -y
conda install curl -y
conda install cmake -y

pip install -U bitsandbytes
pip install -U git+https://github.com/huggingface/transformers.git
pip install -U git+https://github.com/huggingface/peft.git
pip install -U git+https://github.com/huggingface/accelerate.git
pip install datasets
pip install scipy

pip install psycopg2
pip install psycopg2-binary
pip install sentencepiece
pip install tiktoken

pip install autotrain-advanced
`

**Comment by stavsap** [comment]
this is my script to install miniconda 3

curl -sL "https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh" > "Miniconda3.sh"
bash Miniconda3.sh -b 

echo 'export PATH="$PATH:/home/$USER/miniconda3/bin/"' >> .bashrc
echo 'export LD_LIBRARY_PATH="/usr/lib/wsl/lib:/home/$USER/miniconda3/lib/"' >> .bashrc

source .bashrc 
conda init bash
source .bashrc
nvidia-smi

**Comment by helloworld4774** [comment]
having the same issue on windows what can i do here? thanks

**Comment by SerenityNrrd** [comment]
I have tried  to install this a few times on windows and WSL without any luck. Hopefully soon there will be an easy install method eventually.

**Comment by abhishekkrthakur** [comment]
Hopefully fixed tomorrow :)

**Comment by SerenityNrrd** [comment]
Is there any function that you have or that is out there that can help create datasets? specially one that is what is needed for the training?
